plugins/pdr_deterministic_plugin/ufm_sim_web_service/isolation_mgr.py

```diff
@@ -485,35 +546,48 @@ def run_telemetry_get_port(self):
         endpoint_port = self.ufm_client.dynamic_session_get_port(Constants.PDR_DYNAMIC_NAME)
         return endpoint_port
 
+    def restart_telemetry_session(self):
+        self.logger.info("Restarting telemetry session")
+        self.ufm_client.stop_dynamic_session(Constants.PDR_DYNAMIC_NAME)
+        time.sleep(self.dynamic_wait_time)
+        endpoint_port = self.run_telemetry_get_port()
+        return endpoint_port
+
     def main_flow(self):
         # sync to the telemetry clock by blocking read
         self.logger.info("Isolation Manager initialized, starting isolation loop")
         self.get_ports_metadata()    
         self.logger.info("Retrieved ports metadata")
         endpoint_port = self.run_telemetry_get_port()
         self.logger.info("telemetry session started")
+        dynamic_telemetry_unresponsive_count = 0
         while(True):
             try:
                 t_begin = time.time()
                 self.get_isolation_state()
                 self.logger.info("Retrieving telemetry data to determine ports' states")
-                issues = self.read_next_set_of_high_ber_or_pdr_ports(endpoint_port)
+                try:
+                    issues = self.read_next_set_of_high_ber_or_pdr_ports(endpoint_port)
+                except DynamicTelemetryUnresponsive as e:
```

Comment:
no error message ?
what for you need e?

Reply:
it is used for the dynamic telemetry health mechanism, log messages are dealt when reaching that point